### ISSUE TITLE
Sliding effect when using 'prev' and 'next' methods

### ISSUE THREAD

**Issue by antoine92190** [fullcalendar/fullcalendar#3666]
Hi Arshaw,

I've been using your plugin for almost a year on my mobile app, thanks a lot for your work! 👍 

I would love to have a sliding effect when swiping with my finger to navigate between views (day, week and month).

I already managed to make the swiping with finger work along with 'prev' and 'next' methods.
However, I had no success when trying to implement the sliding animation when doing so.
I believe it feels more natural on mobile to have this animation.

Is there any way to implement that in your plugin?

Thanks again!

**Comment by arshaw** [comment]
I agree. Will file this under the "responsive" improvements. It is a goal of mine to get this to work at some point. No "quick win" for this.

**Comment by antoine92190** [comment]
Great! Thanks for considering it.
Do you plan on implementing that in 2017 or later is more likely?

**Comment by Simbaclaws** [comment]
@antoine92190 How exactly did you get swiping to work in the first place? I've got mine only to work by adding a extra div and adding the (swipe) gesture to it in typescript, if I put that div over my calendar I can indeed swipe it but any other controls are then disabled because the div is on top of it... How did you get yours to work the way it does? Simply adding the (swipe) gesture to my calendar doesn't work... I'm using ionic and angular.

**Comment by antoine92190** [comment]
The (swipe) gesture is working well above the calendar on iOS, but not on Android.
Not sure why.
Let me know if the swipe gesture works for you on iOS, if not I will share some code.

**Comment by antoine92190** [comment]
And make sure to :thumbsup: my first comment, so this feature is upvoted :wink:

**Comment by MiguelQueiroz** [comment]
Sliding the calendar could trigger the prev / next days. I could do this using a swipe slider using an infinite tab swiper, where the central tab would be day 0, and the +1 or -1 to the sides. Then when we move to a next slide, we move the other to the front, and make that calendar instance to change the day +1 and -1 day .. well for me it sort of works.

**Comment by antoine92190** [comment]
@MiguelQueiroz Sounds good!
Would you mind sharing the code showcasing how you managed that?

**Comment by jassiwaliaVC** [comment]
@MiguelQueiroz can u share the code for the same

**Comment by Shtibel** [comment]
@MiguelQueiroz please share your code

**Comment by saridakis** [comment]
After a lot of try and error code, I think I have found something that may be ok ...  If anyone could try to test it and reply the results back to us, would be great... As you can see instead of 'Swipe' event, I used 'Pan' events ... If you have any better solution, please advice!!!!

<ion-content fullscreen 
  (panLeft)="onPanLeft($event)" 
  (panRight)="onPanRight($event)"  
  (panend)="onPanEnd($event)"
  (datesRender)="datesRender($event)" >

<full-calendar #fullcalendar </full-calendar>
</ion-content>

And on ts file:

 lastPanAction: any;

  onPanLeft($event) {
    this.lastPanAction = $event;    
  }

  onPanRight($event) {
    this.lastPanAction = $event;    
  }

  onPanEnd($event) {
    const fcApi = this.fullcalendar.getApi();
    switch(this.lastPanAction.additionalEvent) {
      case 'panright':
        fcApi.prev();
        break;
      case 'panleft':
        fcApi.next();
        break;
    }
  }

  datesRender($event) {
    console.log($event.view.activeStart, $event.view.activeEnd, $event);
    this.RefreshPage();
  }

**Comment by maykivets** [comment]
@saridakis 
Thank you for the suggested solution. It works. I think it needs an additional check if the gesture did not take to long and it was not too short, e.g.:

`if ($event.deltaTime < 500 && Math.abs($event.deltaX) > 150) {
      switch(this.lastPanAction.additionalEvent) {
        case 'panright':
          fcApi.prev();
          break;
        case 'panleft':
          fcApi.next();
          break;
      }
    }`

**Comment by jamesamuel** [comment]
Are you still planning on implementing this feature?

**Comment by anmol1995-ionic** [comment]
I already managed to make the swiping with finger work along with 'prev' and 'next' methods.

> Hi Arshaw,
> 
> I've been using your plugin for almost a year on my mobile app, thanks a lot for your work! 👍
> 
> I would love to have a sliding effect when swiping with my finger to navigate between views (day, week and month).
> 
> I already managed to make the swiping with finger work along with 'prev' and 'next' methods.
> However, I had no success when trying to implement the sliding animation when doing so.
> I believe it feels more natural on mobile to have this animation.
> 
> Is there any way to implement that in your plugin?
> 
> Thanks again!

how you have done this can you share wit me?

**Comment by MeijerM1** [comment]
Is there any news or activity on this?

**Comment by scruff311** [comment]
> The (swipe) gesture is working well above the calendar on iOS, but not on Android. Not sure why. Let me know if the swipe gesture works for you on iOS, if not I will share some code.

@antoine92190 i too was able to get Swiper working with FullCalendar for prev/next functionality on iOS, but like you, I've found that this does not work on Android. With a little digging, it seems the `touchEnd` event does not fire on Android when you lift your finger from the screen. It gets fired on your subsequent tap of the screen.

It is something specific to the combination of Swiper and FullCalendar. If you load these Codesandoxes on an Android device, you'll see that [Swiper by itself](https://codesandbox.io/s/swiper-default-forked-37443f?file=/index.html) works fine, but when [added to FullCalendar](https://codesandbox.io/s/determined-snow-6mrl6z?file=/src/index.js) it does not.

**Comment by ceddup** [comment]
Adding this to the html page hosting the fullcalendar does the trick. 

<script>
// #3666 fix : swipe on mobile to go next or previous https://github.com/fullcalendar/fullcalendar/issues/3666
let touchStartX = 0;
let touchStartY = 0;

document.addEventListener('touchstart', e => {
  touchStartX = e.touches[0].clientX;
  touchStartY = e.touches[0].clientY;
}, { passive: true });

document.addEventListener('touchend', e => {
  const dx = e.changedTouches[0].clientX - touchStartX;
  const dy = e.changedTouches[0].clientY - touchStartY;

  if (Math.abs(dx) > 50 && Math.abs(dx) > Math.abs(dy)) {
    if (dx > 0) {
      //console.log('Swipe right detected');
	  $('.fc-next-button').click();
    } else {
      //console.log('Swipe left detected');
	  $('.fc-prev-button').click();
    }
  }
}, { passive: true });
</script>
